src/python/grpcio_tests/tests/reflection/_reflection_client_test.py

```diff
@@ -0,0 +1,147 @@
+        self.assertEqual(
+            reflection.SERVICE_NAME, "grpc.reflection.v1alpha.ServerReflection"
+        )
+
+    def testFindFile(self):
+        file_name = _PROTO_FILE_NAME
+        file_desc = self.desc_pool.FindFileByName(file_name)
+        self.assertEqual(file_name, file_desc.name)
+        self.assertEqual(_PROTO_PACKAGE_NAME, file_desc.package)
+        self.assertEqual("proto3", file_desc.syntax)
+        self.assertIn("TestService", file_desc.services_by_name)
+
+        file_name = _EMPTY_PROTO_FILE_NAME
+        file_desc = self.desc_pool.FindFileByName(file_name)
+        self.assertEqual(file_name, file_desc.name)
+        self.assertEqual(_PROTO_PACKAGE_NAME, file_desc.package)
+        self.assertEqual("proto3", file_desc.syntax)
+        self.assertIn("Empty", file_desc.message_types_by_name)
+
+    def testFindFileError(self):
+        with self.assertRaises(KeyError):
+            self.desc_pool.FindFileByName(_INVALID_FILE_NAME)
+
+    def testFindMessage(self):
+        message_name = _EMPTY_PROTO_SYMBOL_NAME
+        message_desc = self.desc_pool.FindMessageTypeByName(message_name)
+        self.assertEqual(message_name, message_desc.full_name)
+        self.assertTrue(message_name.endswith(message_desc.name))
+
+    def testFindMessageError(self):
+        with self.assertRaises(KeyError):
+            self.desc_pool.FindMessageTypeByName(_INVALID_SYMBOL_NAME)
+
+    def testFindServiceFindMethod(self):
+        service_name = self._SERVICE_NAMES[0]
+        service_desc = self.desc_pool.FindServiceByName(service_name)
+        self.assertEqual(service_name, service_desc.full_name)
+        self.assertTrue(service_name.endswith(service_desc.name))
+        file_name = _PROTO_FILE_NAME
+        file_desc = self.desc_pool.FindFileByName(file_name)
+        self.assertIs(file_desc, service_desc.file)
+
+        method_name = "EmptyCall"
+        self.assertIn(method_name, service_desc.methods_by_name)
+
+        method_desc = service_desc.FindMethodByName(method_name)
+        self.assertIs(method_desc, service_desc.methods_by_name[method_name])
+        self.assertIs(service_desc, method_desc.containing_service)
+        self.assertEqual(method_name, method_desc.name)
+        self.assertTrue(method_desc.full_name.endswith(method_name))
+
+        empty_message_desc = self.desc_pool.FindMessageTypeByName(
+            _EMPTY_PROTO_SYMBOL_NAME
+        )
+        self.assertEqual(empty_message_desc, method_desc.input_type)
+        self.assertEqual(empty_message_desc, method_desc.output_type)
+
+    def testFindServiceError(self):
+        with self.assertRaises(KeyError):
+            self.desc_pool.FindServiceByName(_INVALID_SYMBOL_NAME)
+
+    def testFindServiceFindMethod(self):
+        service_name = self._SERVICE_NAMES[0]
+        service_desc = self.desc_pool.FindServiceByName(service_name)
+
+        with self.assertRaises(KeyError):
+            service_desc.FindMethodByName(_INVALID_SYMBOL_NAME)
```

Comment:
nit: Put this as `testFindServiceFindMethodError`?

Reply:
Done. Moved to 2 separate test cases (`testFindServiceError`, `testFindMethodError`).